src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -365,7 +369,7 @@ public LowLevelTransaction(StorageEnvironment env, long id, TransactionPersisten
                     // for write transactions, we can use the current one (which == null)
                     _scratchPagerStates = scratchPagerStates;
 
-                    _state = env.State.Clone();
+                    _state = env.State;
```

Comment:
Actually no, because that branch is when we are in ReadOnly mode. Either way, the code is confusing. I know because I had to recheck the code several time when writing and again when you mentioned it. So I am refactoring that to make it more accessible.

Reply:
No, that branch is when you're in a write transaction so we're not in a ReadOnly mode. Anyway I see it got fixed.